submarine-workbench/workbench-server/src/test/java/org/apache/submarine/server/WorkbenchGitHubServerTest.java

```diff
@@ -0,0 +1,266 @@
+import org.eclipse.jgit.transport.PushResult;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.apache.submarine.database.utils.HttpRequestUtil.sendHttpRequest;
+
+public class WorkbenchGitHubServerTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WorkbenchGitHubServerTest.class);
+  private static final String OWNER = "submarine-thirdparty";
+  private static final String REPO = "submarine_git_test";
+  WorkbenchGitHubServer workbenchGitHubServer = new WorkbenchGitHubServer();
+  private static final String REMOTE_PATH = "https://github.com/" + OWNER + "/" + REPO + ".git";
+  private static final String BRANCHNAME = "Branch-1";
+  private static final String LOCALPATH =
+      WorkbenchGitHubServerTest.class.getResource("").toString().substring(6) + REPO;
+
+  @After
+  public void restoreAllOperations() {
+    String token = System.getenv("gitToken");
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
+      sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO + "/subscription",
+          map, null, "DELETE");
+      sendHttpRequest("https://api.github.com/user/starred/" + OWNER + "/" + REPO + "",
+          map, null, "DELETE");
+
+      String result = sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO
+          + "/branches", map, null, "GET");
+      Gson gson = new Gson();
+      List<Map<String, String>> list = gson.fromJson(result, List.class);
+      for (Map<String, String> stringStringMap : list) {
+        if (BRANCHNAME.equals(stringStringMap.get("name"))) {
+          sendHttpRequest("https://api.github.com/repos/" + OWNER + "/" + REPO
+              + "/git/refs/heads/" + BRANCHNAME, map, null, "DELETE");
+        }
+      }
+      LOG.info("LOCALPATH: {}", LOCALPATH);
+      deleteDirectory(LOCALPATH);
+    }
+  }
+
+  public void deleteDirectory(String dir) {
+    File dirFile = new File(dir);
+    File[] files = dirFile.listFiles();
+    if (files != null) {
+      for (File file : files) {
+        if (file.isFile()) {
+          new File(file.getAbsolutePath()).delete();
+        } else if (file.isDirectory()) {
+          deleteDirectory(file.getAbsolutePath());
+        }
+      }
+      dirFile.delete();
+    }
+  }
+
+  @Test
+  public void addWatching() {
+    String token = System.getenv("gitToken");
+    if (token == null) {
+      LOG.warn("Token not set!");
+      return;
+    } else {
+      Map<String, String> map = new HashMap<>();
+      map.put("Authorization", "token " + token);
+      LOG.info("gitToken: {}", token);
```

Comment:
For security reasons, token cannot be printed.

Reply:
Done.